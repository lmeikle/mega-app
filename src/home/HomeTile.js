import React from 'react';
import { Link } from 'react-router-dom';
import { string } from 'prop-types';
import bankingImage from './images/banking.png';
import formsImage from './images/forms.png';
import reactImage from './images/react.png';
import newsImage from './images/news.png';
import responsiveImage from './images/responsive.png';
import miscImage from './images/misc.png';
import uiToolsImage from './images/uitools.png';
import styled, { css } from 'styled-components';

const HomeTileWrapper = styled.div`
  box-sizing: border-box;
  display: flex;
  justify-content: center;
  align-items: flex-end;
  width: 150px;
  height: 150px;
  padding: 5px;
  margin: 10px;
  background-color: #fafafa;
  border: 2px solid ${props => props.theme.primary};
  border-radius: 10px;
  background-size: 80%;
  background-repeat: no-repeat;
  background-position: center 10px;
  font-weight: bold;
  font-size: 20px;

  &:hover {
    background-color: #ffffe0;
  }

  div {
    color: ${props => props.theme.primary};
  }

  ${props => {
    switch (props.name) {
      case 'Banking': {
        return bankingMixin;
      }
      case 'Forms': {
        return formsMixin;
      }
      case 'React Playground': {
        return reactPlaygroundMixin;
      }
      case 'News': {
        return newsMixin;
      }
      case 'Responsive': {
        return responsiveMixin;
      }
      case 'Misc': {
        return miscMixin;
      }
      case 'UI Tools': {
        return uitoolsMixin;
      }
      default:
        break;
    }
  }};
`;

const bankingMixin = css`
  background-image: url(${bankingImage});
  background-size: 60%;
  background-position: center 15px;
`;

const formsMixin = css`
  background-image: url(${formsImage});
  background-size: 70%;
  background-position: 30px 10px;
`;

const reactPlaygroundMixin = css`
  background-image: url(${reactImage});
  background-size: 110%;
  background-position: center 0px;
`;

const newsMixin = css`
  background-image: url(${newsImage});
  background-size: 65%;
  background-position: center 15px;
`;

const responsiveMixin = css`
  background-image: url(${responsiveImage});
  background-position: center 4px;
`;

const miscMixin = css`
  background-image: url(${miscImage});
  background-size: 65%;
  background-position: center 15px;
`;

const uitoolsMixin = css`
  background-image: url(${uiToolsImage});
  background-size: 65%;
  background-position: center 15px;
`;

const HomeTile = ({ path, name }) => {
  return (
    <Link to={{ pathname: path }}>
      <HomeTileWrapper name={name}>
        <div>{name}</div>
      </HomeTileWrapper>
    </Link>
  );
};

HomeTile.propTypes = {
  path: string.isRequired,
  name: string.isRequired,
  className: string.isRequired
};

HomeTile.defaultProps = {};

export default HomeTile;
