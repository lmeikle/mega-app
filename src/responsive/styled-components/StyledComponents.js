import React, { Component } from 'react';
import styled from 'styled-components';

const Wrapper = styled.div`
  display: flex;
  justify-content: center;
  flex-wrap: wrap;

  @media only screen and (max-width: 768px) {
    flex-direction: column;
  }
`;

const Item = styled.div`
  flex-grow: 1;
`;

export default class StyledComponents extends Component {
  render() {
    return (
      <Wrapper>
        <Item>
          <h3>Div 1</h3>
        </Item>
        <Item>
          <h3>Div 2</h3>
        </Item>
        <Item>
          <h3>Div 3</h3>
        </Item>
      </Wrapper>
    );
  }
}
