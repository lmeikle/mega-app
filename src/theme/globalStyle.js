import styled, { css, injectGlobal } from 'styled-components';

injectGlobal`
  @import url('https://fonts.googleapis.com/css?family=Montserrat|Roboto');
  
  body {
    padding: 0;
    margin: 0;
    font-family: Roboto, sans-serif;
  }

  #root {
    position: absolute;
    width: 100%;
    height: 100%;
  }

  #modal-root {
    display: none;
    position: absolute;
    width: 100%;
    height: 100%;
    background: rgba(1, 1, 1, 0.3);
  }
`;

// https://medium.com/styled-components/styled-components-getting-started-c9818acbcbbd
export const theme = {
  primary: '#002742',
  secondary: '#4183C4'
};

// some global buttons I could use
export const MegaAppButton = styled.button`
  @import url('https://fonts.googleapis.com/css?family=Jim+Nightshade');
  font-family: 'Jim Nightshade', cursive;

  font-size: 1rem;
  border-radius: 5px;
  padding: 0.25rem 1rem;
  margin: 0 1rem;
  background: transparent;
  color: ${props => props.theme.primary};
  border: 2px solid ${props => props.theme.primary};
  ${props =>
    props.primary &&
    css`
      background: ${props => props.theme.primary};
      color: white;
    `};
`;

export const MegaAppBigButton = MegaAppButton.extend`
  height: 3rem;
  font-size: 2rem;
  width: 40vw;
  border-radius: 30px;
`;
