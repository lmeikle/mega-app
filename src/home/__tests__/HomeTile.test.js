import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';
import { MemoryRouter } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';
import { theme } from '../../theme/globalStyle';
import HomeTile from '../HomeTile';

let props = {
  path: '/banking',
  name: 'Banking',
  className: 'banking'
};

describe('HomeTile', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(
      <ThemeProvider theme={theme}>
        <MemoryRouter>
          <HomeTile {...props} />
        </MemoryRouter>
      </ThemeProvider>
    );
  });

  test('renders correctly', () => {
    let homeTileWrapper = wrapper.find('HomeTile__HomeTileWrapper div');
    let homeTileWrapperJson = toJson(homeTileWrapper);

    expect(homeTileWrapperJson).toMatchSnapshot();
  });

  test('uses the global theme correctly', () => {
    let homeTileWrapper = wrapper.find('HomeTile__HomeTileWrapper div');

    expect(homeTileWrapper).toHaveStyleRule('border', `2px solid ${theme.primary}`);
  });
});
