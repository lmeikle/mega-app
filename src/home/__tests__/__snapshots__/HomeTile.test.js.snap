// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeTile renders correctly 1`] = `
Array [
  .c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  width: 150px;
  height: 150px;
  padding: 5px;
  margin: 10px;
  background-color: #fafafa;
  border: 2px solid #002742;
  border-radius: 10px;
  background-size: 80%;
  background-repeat: no-repeat;
  background-position: center 10px;
  font-weight: bold;
  font-size: 20px;
  background-image: url(banking.png);
  background-size: 60%;
  background-position: center 15px;
}

.c0:hover {
  background-color: #ffffe0;
}

.c0 div {
  color: #002742;
}

<div
    className="c0"
    name="Banking"
  >
    <div>
      Banking
    </div>
  </div>,
  <div>
    Banking
  </div>,
]
`;
